src/StakeRegistry.sol

```diff
@@ -187,14 +137,65 @@ contract StakeRegistry is StakeRegistryStorage {
         }
     }
 
+    /**
+     * @notice Called by the registry coordinator to update an operator's stake for one
+     * or more quorums.
+     *
+     * If the operator no longer has the minimum stake required for a quorum, they are
+     * added to the
+     */
+    function updateOperatorStake(
+        address operator, 
+        bytes32 operatorId, 
+        bytes calldata quorumNumbers
+    ) external onlyRegistryCoordinator returns (uint192) {
+        uint192 quorumsToRemove;
+
+        /**
+         * For each quorum, update the operator's stake and record the delta
+         * in the quorum's total stake.
+         *
+         * If the operator no longer has the minimum stake required to be registered
+         * in the quorum, the quorum number is added to `quorumsToRemove`, which
+         * is returned to the registry coordinator.
+         */
+        for (uint256 i = 0; i < quorumNumbers.length; i++) {
+            uint8 quorumNumber = uint8(quorumNumbers[i]);
+            require(_quorumExists(quorumNumber), "StakeRegistry.updateOperatorStake: quorum does not exist");
+
+            // Fetch the operator's current stake, applying weighting parameters and checking
+            // against the minimum stake requirements for the quorum.
+            (uint96 stakeWeight, bool hasMinimumStake) = _weightOfOperatorForQuorum(quorumNumber, operator);
+
+            // If the operator no longer meets the minimum stake, set their stake to zero and mark them for removal
+            if (!hasMinimumStake) {
+                stakeWeight = 0;
+                quorumsToRemove |= quorumNumber;
```

Comment:
Rather than using a bitmap for `quorumsToRemove` I think it makes more sense to use a bytes array instead here. This is only called in `updateOperators()` in the registryCoordinator which converts it back to a bytes array anyways for the deregistering

Reply:
I hear you, but the additional syntax required to create that dynamic array in `updateOperatorStake` is super annoying.

And the end result is probably similar in gas cost, if i had to guess.

Reply:
Sounds good, yeah probably not huge gas savings